addons/hr_holidays/models/hr_leave.py

```diff
@@ -679,7 +683,10 @@ def create(self, vals_list):
                 if mapped_validation_type[leave_type_id] == 'both':
                     self._check_double_validation_rules(employee_id, values.get('state', False))
 
-        holidays = super(HolidaysRequest, self.with_context(mail_create_nosubscribe=True)).create(vals_list)
+        holidays = super(HolidaysRequest, self.with_context(
+            mail_create_nosubscribe=True,
+            default_holiday_status_id=leave_types and leave_types[0]
```

Comment:
Is it really needed?

PS: could you also change your commit message's title to have more like a sentence? :-D

Reply:
Yep it's necessary + :ok_hand: the title should be more explicit now :D

Reply:
@kba-odoo

Reply:
Oops :see_no_evil:  Why is the `default_holiday_status_id` key needed?

Reply:
After trying to understand why I did it like that, I figure that I was solving the issue in a roundabout way (change the default_get so state is correctly compute) instead of fixing the computation of state itself in our problematic case :man_facepalming:

Reply:
I'm just not sure if the first case should be nested in the second or not because this code validate on creation if the leave request is created by a time-off manager